filmreel/src/cut.rs

```diff
@@ -1,49 +1,186 @@
+use crate::error::Error;
 use crate::utils::ordered_map;
 
 use lazy_static::lazy_static;
 use regex::Regex;
-use serde::de::DeserializeOwned;
 use serde::{Deserialize, Serialize};
-use serde_json::Value;
 use std::collections::HashMap;
+use std::ops::Range;
 
-/// Holds cut variables and their corresonding values stored in a series of key/value pairs.
+/// Holds Cut Variables and their corresonding values stored in a series of key/value pairs.
 ///
 /// [Cut Register](https://github.com/Bestowinc/filmReel/blob/supra_dump/cut.md#cut-register)
 #[derive(Serialize, Deserialize, Default, Debug, PartialEq)]
 pub struct Register<'a> {
     #[serde(serialize_with = "ordered_map", borrow, flatten)]
-    vars: Vars<'a>,
+    vars: Variables<'a>,
 }
 
+const VAR_NAME_ERR: &'static str ="Only alphanumeric characters, dashes, and underscores are permitted in Cut Variable names => [A-za-z_]";
+/// The Register's map of [Cut Variables](https://github.com/Bestowinc/filmReel/blob/supra_dump/cut.md#cut-variable)
+type Variables<'a> = HashMap<&'a str, &'a str>;
+
+#[allow(dead_code)] // FIXME
 impl<'a> Register<'a> {
-    pub fn new() -> Self {
-        Default::default()
-    }
-    pub fn insert(&mut self, key: &'a str, val: &'a str) -> Result<Option<&'a str>, &str> {
+    /// Inserts entry into the Register's Cut Variables/
+    ///
+    /// Returns an Err if the key value is does not consist solely of characters, dashes, and underscores.
+    pub fn insert(&mut self, key: &'a str, val: &'a str) -> Result<Option<&'a str>, Error> {
         lazy_static! {
             // Permit only alphachars dashes and underscores for variable names
             static ref KEY_CHECK: Regex = Regex::new(r"^[A-za-z_]+$").unwrap();
         }
         if !KEY_CHECK.is_match(key) {
-            return Err(
-                "Only alphanumeric characters, dashes, and underscores are permitted in cut variable names: [A-za-z_]");
+            return Err(Error::FrameParsef(VAR_NAME_ERR, key.to_string()));
         }
         Ok(self.vars.insert(key, val))
     }
+
+    /// Gets a reference to the string slice value for the given var name.
+    ///
+    /// [Cut Variable](https://github.com/Bestowinc/filmReel/blob/supra_dump/cut.md#cut-variable)
+    pub fn get_kv(&self, key: &str) -> Option<(&&str, &&str)> {
```

Comment:
minor suggestion: keep the naming consistent for the `get_key_value` function across structs
```suggestion
    pub fn get_key_value(&self, key: &str) -> Option<(&&str, &&str)> {
```